client/state/data-layer/wpcom/activity-log/share/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import { recordTracksEvent, withAnalytics } from 'calypso/state/analytics/actions';
+import { REWIND_ACTIVITY_SHARE_REQUEST } from 'calypso/state/action-types';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+
+const requestShare = ( action ) =>
+	http(
+		{
+			apiVersion: '1',
+			method: 'POST',
+			path: `/activity-log/${ action.siteId }/share/${ action.activityId }`,
+			body: {
+				email: action.email,
+			},
+		},
+		action
+	);
```

Comment:
Here's what an example API request would look like this:
`/wpcom/v2/sites/{BLOG_ID}/activity/{REWIND_ID}/share?email={RECIPIENT_EMAIL}`
Note that it uses REWIND_ID and not ACTIVITY_ID due to Rewind internals.

Reply:
Updated PR to reflect proper API path. Please note that the email field will be urlencoded.

Reply:
Also updated to use rewindId :-)